server/store.py

```diff
@@ -0,0 +1,337 @@
+            daily_room_name: str = None,
+            max_videos: int = None,
+            transcriber: Transcriber = None):
+        self.daily_room_name = daily_room_name
+        self.max_videos = max_videos
+        if not transcriber:
+            # Default to local Whisper model if Deepgram API key is not
+            # specified
+            transcriber = WhisperTranscriber()
+            deepgram_api_key = os.getenv("DEEPGRAM_API_KEY")
+            if deepgram_api_key:
+                transcriber = DeepgramTranscriber()
+        self.transcriber = transcriber
+
+    def query(self, query: str) -> Response:
+        """Queries the existing index, if one exists."""
+        if not self.ready():
+            raise Exception("Index not yet initialized. Try again later")
+        engine = self.index.as_query_engine()
+        response = engine.query(query)
+        return response
+
+    def load_index(self) -> bool:
+        """Attempts to load existing index"""
+        self.update_status(State.LOADING, "Loading index")
+        try:
+            save_dir = get_index_dir_path()
+            vector_store = self.get_vector_store()
+            storage_context = StorageContext.from_defaults(
+                vector_store=vector_store,
+                docstore=SimpleDocumentStore.from_persist_dir(
+                    persist_dir=save_dir),
+                index_store=SimpleIndexStore.from_persist_dir(
+                    persist_dir=save_dir),
+            )
+            index = load_index_from_storage(storage_context)
+            if index is not None:
+                self.index = index
+                self.update_status(
+                    State.READY, "Index loaded and ready to query")
+                return True
+        except FileNotFoundError:
+            print("Existing index not found. Store will not be loaded.")
+        except ValueError as e:
+            print("Failed to load index; collection likely not found", e)
+        self.update_status(State.UNINITIALIZED)
+        return False
+
+    async def initialize_or_update(self, source: Source):
+        """Initializes or updates a vector store from given source"""
+
+        # If index does not already exist, create it.
+        if not self.ready():
+            self.update_status(State.CREATING, "Creating index")
+            try:
+                await self.generate_index(source)
+            except Exception as e:
+                msg = "Failed to create index"
+                print(f"{msg}: {e}", file=sys.stderr)
+                self.update_status(State.ERROR, "Failed to create index")
+                return
+
+        # This will fetch any _new_ recordings
+        # and update the existing index
+        self.update_status(State.UPDATING, "Updating index")
+        try:
+            if source == Source.DAILY:
+                await self.index_daily_recordings()
+            elif source == Source.UPLOADS:
+                await self.generate_upload_transcripts()
```

Comment:
maybe this could be:

```
await self.generate_index(source, False)
```

where `False` is from:

```
    async def generate_index(self, source: Source, create_index: bool):
        """Generates a new index from given source"""
        if source == Source.DAILY:
            await self.index_daily_recordings()
        elif source == Source.UPLOADS:
            await self.generate_upload_transcripts()
        if (create_index):
            self.create_index()
```

Reply:
🤦  Of course